@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -0,0 +1,1268 @@
+import { EarthConstants, GeoCoordinates, ProjectionType } from "@here/harp-geoutils";
+import {
+    DashedLineMaterial,
+    EdgeMaterial,
+    EdgeMaterialParameters,
+    FadingFeature,
+    MapMeshBasicMaterial,
+    SolidLineMaterial
+} from "@here/harp-materials";
+import {
+    ContextualArabicConverter,
+    FontStyle,
+    FontUnit,
+    FontVariant,
+    HorizontalAlignment,
+    TextLayoutStyle,
+    TextRenderStyle,
+    VerticalAlignment,
+    WrappingMode
+} from "@here/harp-text-canvas";
+import { getOptionValue } from "@here/harp-utils";
+import * as THREE from "three";
+
+import { AnimatedExtrusionTileHandler } from "../AnimatedExtrusionHandler";
+import { ColorCache } from "../ColorCache";
+import { createMaterial, getBufferAttribute, getObjectConstructor } from "../DecodedTileHelpers";
+import {
+    createDepthPrePassMesh,
+    isRenderDepthPrePassEnabled,
+    setDepthPrePassStencil
+} from "../DepthPrePass";
+import { MapViewPoints } from "../MapViewPoints";
+import { TextElement } from "../text/TextElement";
+import { DEFAULT_TEXT_DISTANCE_SCALE } from "../text/TextElementsRenderer";
+import { computeStyleCacheId } from "../text/TextStyleCache";
+import { Tile, TileFeatureData } from "../Tile";
+
+// tslint:disable:max-line-length
+/**
+ * The SORT_WEIGHT_PATH_LENGTH constants control how the priority of the labels are computed based
+ * on the length of the label strings.
+ *
+ * Consequently, the [[Technique]]s priority is slightly modified while generating
+ * [[TextElement]]s from the [[DecodedTile]], to get a more meaningful priority and stable results.
+ */
+
+/**
+ * Gives [[TextElement]]s with longer paths a higher priority.
+ */
+const SORT_WEIGHT_PATH_LENGTH = 0.1;
+
+/**
+ * Parameters that control fading.
+ */
+export interface FadingParameters {
+    fadeNear?: number;
+    fadeFar?: number;
+}
+
+/**
+ * Parameters that control fading for extruded buildings with fading edges.
+ */
+export interface PolygonFadingParameters extends FadingParameters {
+    color?: string | number;
+    colorMix?: number;
+    lineFadeNear?: number;
+    lineFadeFar?: number;
+}
+
+/**
+ * Support class to create geometry for a [[Tile]] from a [[DecodedTile]].
+ */
+export class TileGeometryCreator {
+    /**
+     * Cache for named colors.
+     */
+    private static m_colorMap: Map<string, THREE.Color> = new Map();
+
+    /**
+     * Prepare decoded tile for geometry creation.
+     *
+     * @param decodedTile The [[DecodedTile]].
+     */
+    initDecodedTile(_decodedTile: DecodedTile) {
+        //
+    }
+
+    /**
+     * Called after the `Tile` has been decoded.
+     *
+     * @param tile The [[Tile]] to process.
+     * @param decodedTile The [[DecodedTile]].
+     */
+    createAllGeometries(tile: Tile, decodedTile: DecodedTile) {
+        tile.clear();
+        this.preparePois(tile, decodedTile);
+        this.createTextElements(tile, decodedTile);
+        this.addGroundPlane(tile);
```

Comment:
this is also called in this.createObjects, do you mean to add it twice?

Reply:
Of course, just to be sure!

Just kidding... Removed the call to addGroundPlane()  here.